src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -1219,10 +1226,22 @@ static int _xccdf_policy_generate_fix_blueprint(struct oscap_list *rules_to_fix,
 		if (oscap_iterator_has_more(services_disable_it))
 			_write_text_to_fd(output_fd, ",");
 	}
-	_write_text_to_fd(output_fd, "]\n\n");
+	_write_text_to_fd(output_fd, "]\n");
 	oscap_iterator_free(services_disable_it);
 	oscap_list_free(services_disable, free);
 
+	_write_text_to_fd(output_fd, "masked = [");
+	struct oscap_iterator *services_mask_it = oscap_iterator_new(services_mask);
+	while(oscap_iterator_has_more(services_mask_it)) {
+		char *var_line = (char *) oscap_iterator_next(services_mask_it);
+		_write_text_to_fd(output_fd, var_line);
+		if (oscap_iterator_has_more(services_mask_it))
+			_write_text_to_fd(output_fd, ",");
+	}
+	_write_text_to_fd(output_fd, "]\n\n");
+	oscap_iterator_free(services_mask_it);
+	oscap_list_free(services_mask, free);
```

Comment:
This block seems to be a repeating theme. Can we refactor the code to reduce duplication?

Reply:
Sure, can be done.